js/main.js

```diff
@@ -1,26 +1,127 @@
+/* eslint-disable no-unused-vars */
 'use strict';
 
+const AVATAR_INDEX = {
+  min: 1,
+  max: 8,
+};
+
+const LOCATION_X = {
+  min: 35.65000,
+  max: 35.70000,
+};
+
+const LOCATION_Y = {
+  min: 139.70000,
+  max: 139.80000,
+};
+
+const FLOAT_DIGITS = 5;
+
+const PRICE = {
+  min: 100,
+  max: 1000,
+};
+
+const ROOMS = {
+  min: 1,
+  max: 50,
+};
+
+const GUESTS = {
+  min: 1,
+  max: 50,
+};
+
+const TITLES_LIST = [
+  'Rustic Private Cottage with King Bed and Oceanside View.',
+  'Luxury Beach House Perfect for Romantic Weekend Getaway.',
+  'Enjoy Sunsets at Fully Equipped 2BR Condo at the Beach.',
+];
+
+const TYPES_LIST = ['palace', 'flat', 'house', 'bungalow'];
+
+const CHECKIN_TIME_LIST = ['12:00', '13:00', '14:00'];
+
+const FEATURES_LIST = ['wifi', 'dishwasher', 'parking', 'washer', 'elevator', 'conditioner'];
+
+const DESCRIPTIONS_LIST = [
+  'This bright loft is perfect to relax, just up the road from the heart of the city.',
+  'Have a glass of wine while enjoying Netflix on the comfortable sofa after exploring local culture.',
+  '10 minute bus ride to downtown. Super safe and quiet neighborhood.',
+];
+
+const PHOTOS_LIST = [
+  'http://o0.github.io/assets/images/tokyo/hotel1.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel2.jpg',
+  'http://o0.github.io/assets/images/tokyo/hotel3.jpg',
+];
+
+const SIMILAR_LISTINGS_COUNT = 10;
+
 const getRandomIntInclusive = function(min, max) {
   min = Math.ceil(min);
   max = Math.floor(max);
 
-  if (min >= 0 && max >= 0) {
+  if (min >= 0 && max > min) {
     return Math.floor(Math.random() * (max - min + 1) + min);
   } else {
     return false;
   }
-}
-
-getRandomIntInclusive(10.09, 0.09);
-
+};
 
 const getRandomFloatInclusive = function(min, max, floatDigits) {
 
-  if (min >= 0 && max >= 0 && floatDigits >= 0) {
-    return +(Math.random() * (max - min + 1) + min).toFixed(floatDigits);
+  if (min >= 0 && max > min && floatDigits > 0) {
+    return +(Math.random() * (max - min) + min).toFixed(floatDigits);
   } else {
     return false;
   }
+};
+
+const generateRandomValue = (array) => array[getRandomIntInclusive(0, array.length - 1)];
+
+
+const shuffleArray = (array) => {
+  for (let i = array.length - 1; i > 0; i--) {
+    let j = Math.floor(Math.random() * (i + 1));
+    let temp = array[i];
+    array[i] = array[j];
+    array[j] = temp;
+  }
+  return array;
 }
 
-getRandomFloatInclusive(5.098, 3.789, 7);
+const generateRandomList = (shuffleArray) => shuffleArray.slice(getRandomIntInclusive(0, shuffleArray.length));
```

Comment:
Может тебе не пользоваться стрелочными функциями пока тебе сама функция не понятно как работает?
Смотри, тут shuffleArray - это название параметра функции generateRandomList, если переназвать это как
```suggestion
const generateRandomList = (array) => array.slice(getRandomIntInclusive(0, array.length));
```
то всё будет работать.

Reply:
Также, метод slice не изменяет сам массив, поэтому его результат надо _вернуть_ из функции
```suggestion
const generateRandomList = (shuffleArray) => return shuffleArray.slice(getRandomIntInclusive(0, shuffleArray.length));
```

из-за этого у тебя и были undefined в этих местах в similarListings

Reply:
Я, конечно, не помню наизусть какие методы массива изменяют его, а которые делают новый массив из старого. Обычно я смотрю в книжке JavaScript. Подробное руководство, Фленагана, но можно смотреть в MDN. Еще можно сделать так:
```
const generateRandomList = (mns) => {
  console.log('mns', mns);
  mns.slice(getRandomIntInclusive(0, mns.length));
  console.log('mns 2', mns);
};
```

и выясняется, что массив не меняется.

тогда исправляем:
```
const generateRandomList = (mns) => {
  console.log('mns', mns);
  let tns = mns.slice(getRandomIntInclusive(0, mns.length));
  console.log('tns', tns);
};
```
так уже меняется. И теперь остается только вернуть переменную tns, ну или как я тебе выше написала, просто вернуть результат слайса, не используя промежуточную переменную

Reply:
mn - это мои инициалы, tn - это моя сестра, для _временных_ переменных я всегда их использую, тк нет смысла придумывать название для переменной, которую потом удалишь